src/core/tests/Test_PatchInstaller.py

```diff
@@ -189,6 +189,56 @@ def healthstore_writes_helper(self, health_store_id, maintenance_run_id, expecte
         self.assertEqual(expected_patch_version, json.loads(substatus_file_data['formattedMessage']['message'])['patchVersion'])
         runtime.stop()
 
+    def test_raise_if_telemetry_unsupported(self):
+        # Constants.VMCloudType.ARC
+        current_time = datetime.datetime.utcnow()
+        td = datetime.timedelta(hours=1, minutes=2)
+        job_start_time = (current_time - td).strftime("%Y-%m-%dT%H:%M:%S.9999Z")
+        argument_composer = ArgumentComposer()
+        argument_composer.maximum_duration = 'PT1H'
+        argument_composer.start_time = job_start_time
+        runtime = RuntimeCompositor(argument_composer.get_composed_arguments(env_settings={"telemetrySupported": True}), True, Constants.YUM)
+        runtime.set_legacy_test_type('SuccessInstallPath')
+        runtime.patch_installer.raise_if_telemetry_unsupported()
+        runtime.stop()
+
+        runtime = RuntimeCompositor(argument_composer.get_composed_arguments(env_settings={"telemetrySupported": False}), True, Constants.YUM)
+        runtime.set_legacy_test_type('SuccessInstallPath')
+        self.assertRaises(Exception, runtime.patch_installer.raise_if_telemetry_unsupported)
+        runtime.stop()
+
+        runtime = RuntimeCompositor(argument_composer.get_composed_arguments(env_settings={"telemetrySupported": True}), True, Constants.YUM)
+        runtime.set_legacy_test_type('SuccessInstallPath')
+        runtime.patch_installer.lifecycle_manager.get_vm_cloud_type = lambda: Constants.VMCloudType.ARC
+        runtime.patch_installer.raise_if_telemetry_unsupported()
```

Comment:
Is there any assert to check here? The function returns a value or sets something

Reply:
It logs something, but the check here works because it will throw an exception if it fails, and this will fail the overall test. There isn't a way to check for something like `self.assertNotRaises` because an exception will fail the test anyway.